lib/Doctrine/DBAL/Connection.php

```diff
@@ -618,11 +619,7 @@ public function fetchAssociative(string $query, array $params = [], array $types
         try {
             $stmt = $this->executeQuery($query, $params, $types);
 
-            if ($stmt instanceof Result) {
```

Comment:
What if `executeQuery()` is overridden in a custom class and doesn't implement the newly introduced wrapping?

Reply:
You are right, it will break this kind of extension. I propose to use the ensureForwardCompatibleStatement() that should do the work.